src/rendering/batcher/shared/Batcher.ts

```diff
@@ -240,7 +240,7 @@ export class Batcher
         let start = this._batchIndexStart;
 
         let action: BatchAction = 'startBatch';
-        let batch = this._batchPool[this._batchPoolIndex++] || new Batch();
+        let batch = this._batchPool[this._batchPoolIndex++] ||= new Batch();
```

Comment:
Are we worried about this causing a memory leak? A `TextureSource` might never be GC'd because some `BatchTextureArray` in the pool still references it.

I suppose a similar memory leak concern could have `Batcher#_elements`.

Reply:
agree - we have a ticket for this that involves clearing these arrays so references can be released.